### DISCUSSION TITLE
WithDelegateStrategy is executed several times

### DISCUSSION THREAD

**Discussion by mxSoftware3** [Finbuckle/Finbuckle.MultiTenant#839] [Q&A]
Hello, 
I am new in the use of the library, is it possible to use a delegate strategy to get the tenant of the querystring of the request?
I am using the following code but it is executed many times and only the first time it enters it has the tenant, after that the tenant is always null.
builder.Services.AddMultiTenant<MultiTenantInfo>()
     .WithDelegateStrategy(context =>
     {
         var httpContext = context as HttpContext;
         if (httpContext == null)
             return null;

         httpContext.Request.Query.TryGetValue("tenant", out StringValues tenantIdentifier);
         if (tenantIdentifier == "")
             return Task.FromResult<string?>(null);

         return Task.FromResult<string?>(tenantIdentifier.ToString());
     })    
    .WithEFCoreStore<TenantAdminDbContext, MultiTenantInfo>()
    .WithPerTenantAuthentication();
Help !!

**Comment by AndrewTriesToCode** [comment]
Hi that should work but it looks like you won’t have a tenant store configured.

**Reply by mxSoftware3** [reply]
Hello, thank you for answer. The tenants are stored in a sql server DB.

builder.Services.AddDbContext<TenantAdminDbContext>(options =>
            options.UseSqlServer(builder.Configuration.GetConnectionString("TenantAdmin")));
builder.Services.AddDbContext<IdentityServerDbContext>();

builder.Services.AddMultiTenant<MultiTenantInfo>()
     .WithDelegateStrategy(context =>
     {
         var httpContext = context as HttpContext;
         if (httpContext == null)
             return null;

         httpContext.Request.Query.TryGetValue("tenant", out StringValues tenantIdentifier);
         if (tenantIdentifier == "")
             return Task.FromResult<string?>(null);

         return Task.FromResult<string?>(tenantIdentifier.ToString());
     })    
    .WithEFCoreStore<TenantAdminDbContext, MultiTenantInfo>()
    .WithPerTenantAuthentication();

Openid connect is being used in the same solution as an identity server and it is when invoking from the angular client that it enters the server several times. The first time it gets to the delegate it has the correct tenant value but it re-enters the delegate and the tenant value is empty.
